pyanaconda/ui/helpers.py

```diff
@@ -104,7 +108,14 @@ def _clean_hdd_iso(self):
         This means unmounting the partition and unprotecting it,
         so it can be used for the installation.
         """
-        if self.data.method.method == "harddrive" and self.data.method.partition:
+        payload_proxy = get_payload()
```

Comment:
Please, use `self.payload`, `self.payload.proxy`, `self.payload.get_source_proxy` and `set_source`.